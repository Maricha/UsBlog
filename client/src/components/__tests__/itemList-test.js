import React from 'react';
import { shallow } from 'enzyme'; 
import ItemList from '../itemList';

describe('IemList Component', () => {
  it('renders correctly', () => {
    const posts = [
      {
        id: 1,
        title: "title1"
      },
      {
        id: 2,
        title: "title2"
      }
    ]

    const firstPost = {
      id: 3,
      title: 'title3'
    }

    const secondPost = {
      id: 4,
      title: 'title4'
    }

    const component = shallow(<ItemList posts={posts} firstPost={firstPost} secondPost={secondPost} />);
    expect(component).toMatchSnapshot();
  })
});