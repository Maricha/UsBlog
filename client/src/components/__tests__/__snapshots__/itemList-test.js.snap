// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IemList Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Component)
    firstPost={
      Object {
        "id": 3,
        "title": "title3",
      }
    }
    posts={
      Array [
        Object {
          "id": 1,
          "title": "title1",
        },
        Object {
          "id": 2,
          "title": "title2",
        },
      ]
    }
    secondPost={
      Object {
        "id": 4,
        "title": "title4",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "mainStyle": "Component-mainStyle-1",
      },
      "firstPost": Object {
        "id": 3,
        "title": "title3",
      },
      "posts": Array [
        Object {
          "id": 1,
          "title": "title1",
        },
        Object {
          "id": 2,
          "title": "title2",
        },
      ],
      "secondPost": Object {
        "id": 4,
        "title": "title4",
      },
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "type": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "mainStyle": "Component-mainStyle-1",
        },
        "firstPost": Object {
          "id": 3,
          "title": "title3",
        },
        "posts": Array [
          Object {
            "id": 1,
            "title": "title1",
          },
          Object {
            "id": 2,
            "title": "title2",
          },
        ],
        "secondPost": Object {
          "id": 4,
          "title": "title4",
        },
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
